_posts/2020-05-15-guard-duty-orgs.md

```diff
@@ -0,0 +1,246 @@
+---
+layout: post
+title: Rolling Out Guard Duty Organisations
+date: 2020-05-15
+tags: aws guard-duty organisations landing-zone
+author: Simon Bracegirdle
+image: /img/gd_logo.png
+---
+
+<center><img src="/img/gd_logo.png" /></center>
+<br/>
+
+If you're responsible for setting up AWS accounts in your organisation you might be wondering how to ensure a secure and best practice multi-account setup. AWS's threat detection service, [Guard Duty](https://aws.amazon.com/guardduty/), is one of the tools that you should consider including as part of that stack.
+
+In this post we'll go over how Guard Duty can benefit you, what the new "Organisations" feature does and how we can switch it on for accounts in bulk.
+
+
+# What is Guard Duty? How does it work?
+
+Guard Duty automatically monitors resources in your AWS account for potential threats, vulnerabilities and can detect compromised resources. For example, it can detect; IAM privilege escalation, unusual network traffic, compromised EC2 instances and denial of service attacks.
+
+A full list of finding types can be found in the [user guide](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_finding-types-active.html).
+
+When Guard Duty discovers a potential problem, it issues a security finding. Using CloudWatch events, these findings can then be forwarded to an SNS topic to notify administrators by email, or kick off some other automated response such as a precautionary termination of an EC2 instance.
+
+For example, this is what a finding looks like in the AWS web console (NOTE; instance IDs, image IDs and IP addresses are fake):
+
+![Guard Duty finding example](/img/gd_finding_example.png)
+
+# How can I enable it?
+
+Guard Duty is fairly easy to enable for accounts on a case-by-case basis with a single click from the web console. However, if you would like to enable it for [all AWS Organisations accounts en masse](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_accounts.html) and aggregate the security findings into a central account, that's when things can be trickier.
+
+To achieve this, we can make use of the new [Organisations](https://aws.amazon.com/about-aws/whats-new/2020/04/amazon-guardduty-simplifies-multi-account-threat-detection-with-support-for-aws-organizations/) feature in Guard Duty. With this approach a Guard Duty master account needs to be elected, which gives that account permission to add other Organisations accounts as members. There's also an "auto-enable" feature that automatically switches on Guard Duty for new accounts added later on.
+
+Before this feature, invitations would need to be sent from the master account to add accounts as members. Those accounts would then need to accept those invitations. This was tricky to automate as it would require complex cross-account step functions to orchestrate the parts of that process.
```

Comment:
_This was tricky to automate as it would require complex cross-account orchestration to coordinate the process_

There are many ways to do it; the mention of step functions is a potentially distracting detail.